kvbc/include/categorization/details.h

```diff
@@ -27,6 +27,31 @@ namespace concord::kvbc::categorization::detail {
 
 using Buffer = std::vector<std::uint8_t>;
 
+struct TaggedVersion {
+  // The high bit contains a flag indicating whether the key was deleted or not.
+  TaggedVersion(uint64_t masked_version) {
+    deleted = 0x80000000 & masked_version;
+    version = 0x7FFFFFFF & masked_version;
```

Comment:
The masks are 32 bits :) 
I think we just need to use ```TaggedVersion``` as CMF message without using the upper bit as flag.

Reply:
Great catch, thanks @cloudnoize , I've changed to 64 bits in my commit!

In terms of TaggedVersion, as discussed offline, having 1 byte more is not optimal, but it doesn't encode any logic in the message itself and other tools need not be aware of it. Since this byte will be mostly zeroes, maybe it won't have a big impact on storage. Just some thoughts.